rest/specs/src/main/filtered-resources/openapi/openapi.yaml

```diff
@@ -194,32 +194,18 @@ components:
       description: Transformation request.
       type: object
       required:
-        - id
-        - stagedLocation
-        - productLocation
+        - location
         - mimeType
         - bytes
-        - callbackUrl
       properties:
-        id:
-          type: string
-          description: The unique identifier for a request.
-          minLength: 1
-          maxLength: 255
-          pattern: '^[a-zA-Z0-9]*$'
-          example: 30f14c6c1fc85cba12bfd093aa8f90e3
-        stagedLocation:
-          type: string
-          description: The temporary location of the ingested product.
-          minLength: 1
-          maxLength: 2048
-          example: https://bucket.s3.amazonaws.com
-        productLocation:
+        location:
           type: string
-          description: The final location of the ingested product.
+          description: >
+            The location of the ingested product. This URL must support HTTP GET at location/{productId} to retrieve
+            the product and HTTP PUT with attachment at location/{productId}/{metadataType} to store metadata.
```

Comment:
❓ Why are the `/{productId}` and `location/{productId}/{metadataType}` parts included here? Does it matter what the location URI looks like? Should this mention metadata at all?

Reply:
Transformation owns the definition of the endpoint that the location URI must support.  Whoever calls us, must implement the endpoint to our specifications.

Reply:
Why is it important to say anything more than that a product can be retrieved at the URI? Why do you have to mention the metadata and the endpoint specifications of the retrieve URI?

Reply:
If someone were to use this API and send us a message, how would they know how to implement the endpoint that we are going to call if we didn't include this information?  If we didn't specify that it must support a GET and a PUT at those context paths, what would they implement?

Reply:
Oops. I forgot that Transformation Service needs to know how to store metadata 🤦‍♀

Can you update the description to say `location` instead of `location/{productId}`? All you need to know is that you can GET from `<location>` and POST to `<location>/{metadataType}`. Or, alternatively, can you reference the mis-api spec in this description?

Reply:
Emily is correct. The endpoint that receives the POST request is responsible for generating the ID. That endpoint will respond with the location URL that includes the new unique ID.

Reply:
I suspect that ultimately there will be multiple clients of the MIS API. They potentially have different needs than the Transform service. An API should be created in collaboration with its consumers to make sure their needs are met. The team that creates the API is accountable for it and responsible for it. The CDR team has to own the API for that to be possible.

Reply:
Should we then document the store API separately as a different API and simply referenced here that the provifded product URL must support that protocol?

Reply:
Should we then document the store API separately as a different API and simply referenced here that the provided product URL must support that protocol? It has to be documented somewhere and I would say it is not ingest that dictate it but more ingest that is a client of it.

Reply:
Transformation is the one that needs a URI to `GET` the resource and `PUT` new sub-resources under, so it has to be the one defining how it will use that endpoint. Otherwise it would have to know what client submitted the transformation request and how it needs to interact with it. Or it would have to couple itself to the MIS API, which is something we want to avoid to ensure that the Transformation service remains reusable for other purposes in the future.
The way I see this is that we are defining a resource URI structure that can be implemented by any service that needs to receive the output of a transformation request.
MIS can decide to use that same resource URI structure for other clients if it wants to, but it doesn't have to.